sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/FluxAutoComplete.java

```diff
@@ -0,0 +1,130 @@
+    private final Function<ServiceBusReceivedMessageContext, Mono<Void>> onAbandon;
+    private final ClientLogger logger = new ClientLogger(FluxAutoComplete.class);
+
+    FluxAutoComplete(Flux<? extends T> upstream, Function<ServiceBusReceivedMessageContext, Mono<Void>> onComplete,
+        Function<ServiceBusReceivedMessageContext, Mono<Void>> onAbandon) {
+        super(upstream);
+        this.onComplete = Objects.requireNonNull(onComplete, "'onComplete' cannot be null.");
+        this.onAbandon = Objects.requireNonNull(onAbandon, "'onAbandon' cannot be null.");
+    }
+
+    /**
+     * Invoked when a downstream subscriber is interested in objects published from this operator.
+     *
+     * @param coreSubscriber The subscriber interested in the published items.
+     */
+    @Override
+    public void subscribe(CoreSubscriber<? super T> coreSubscriber) {
+        Objects.requireNonNull(coreSubscriber, "'coreSubscriber' cannot be null.");
+
+        final AutoCompleteSubscriber<T> subscriber =
+            new AutoCompleteSubscriber<T>(coreSubscriber, onComplete, onAbandon, logger);
+
+        source.subscribe(subscriber);
+    }
+
+    static final class AutoCompleteSubscriber<T extends ServiceBusReceivedMessageContext> extends BaseSubscriber<T> {
+        private final CoreSubscriber<? super T> downstream;
+        private final Function<ServiceBusReceivedMessageContext, Mono<Void>> onComplete;
+        private final Function<ServiceBusReceivedMessageContext, Mono<Void>> onAbandon;
+        private final ClientLogger logger;
+
+        AutoCompleteSubscriber(CoreSubscriber<? super T> downstream,
+            Function<ServiceBusReceivedMessageContext, Mono<Void>> onComplete,
+            Function<ServiceBusReceivedMessageContext, Mono<Void>> onAbandon, ClientLogger logger) {
+            this.downstream = downstream;
+            this.onComplete = onComplete;
+            this.onAbandon = onAbandon;
+            this.logger = logger;
+        }
+
+        @Override
+        protected void hookOnSubscribe(Subscription subscription) {
+            logger.info("Subscription received. Subscribing downstream. {}", subscription);
+            downstream.onSubscribe(this);
+            requestUnbounded();
+        }
+
+        @Override
+        protected void hookOnNext(T value) {
+            final ServiceBusReceivedMessage message = value.getMessage();
+            final String sequenceNumber = message != null ? String.valueOf(message.getSequenceNumber()) : "n/a";
+
+            logger.verbose("Passing message downstream. sequenceNumber[{}]", sequenceNumber);
+
+            try {
+                downstream.onNext(value);
+                applyWithCatch(onComplete, value, "complete");
+            } catch (Exception e) {
+                logger.error("Error occurred processing message. Abandoning. sequenceNumber[{}]",
+                    sequenceNumber, e);
```

Comment:
should this be info , since we are taking care of this for user, It should not create some kind of alarm for them.